En este documento vamos a realizar un análisis descriptivo de las columnas numéricas y categóricas de nuestros dataframe. 

In [82]:
# Importamos las librerías a utilizar

import pandas as pd # type: ignore
import numpy as np # type: ignore
from src import funciones_soporte as fs
from pandas.api.types import is_datetime64_any_dtype as is_datetime # type: ignore

In [14]:
# Indicamos que queremos que nos enseñe por pantalla todas las columnas de los DataFrame que vamos a cargar
pd.set_option('display.max_columns', None)

In [15]:
datos_brasil = pd.read_pickle("datos/datos_brasil.pkl")
print(datos_brasil.shape)
datos_brasil.head(2)

(1026299, 16)


CÓDIGO ÓRGÃO SUPERIOR       NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
0                  63000  Advocacia-Geral da União         63000   
1                  63000  Advocacia-Geral da União         63000   

                                          NOME ÓRGÃO  CÓDIGO UNIDADE GESTORA  \
0  Advocacia-Geral da União - Unidades com víncul...                  110060   
1  Advocacia-Geral da União - Unidades com víncul...                  110060   

                            NOME UNIDADE GESTORA CATEGORIA ECONÔMICA  \
0  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU  Receitas Correntes   
1  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU  Receitas Correntes   

              ORIGEM RECEITA                                ESPÉCIE RECEITA  \
0  Outras Receitas Correntes  Bens, Direitos e Valores Incorporados ao Patr   
1  Outras Receitas Correntes    Indenizações, restituições e ressarcimentos   

                                    DETALHAMENTO  VALOR PREVISTO ATUALIZADO  \
0  REC.DIVIDA ATIVA NAO TRIBUTARIA DE OUTRAS REC                        0.0   
1   RECUPERACAO DE DESPESAS DE EXERC. ANTERIORES                        0.0   

   VALOR LANÇADO  VALOR REALIZADO  PERCENTUAL REALIZADO DATA LANÇAMENTO  \
0            0.0          1297.13                   0.0      2013-12-31   
1            0.0      26666621.42                   0.0      2013-12-31   

   ANO EXERCÍCIO  
0           2013  
1           2013

In [16]:
#datos_brasil.drop(["CÓDIGO ÓRGÃO SUPERIOR", "CÓDIGO ÓRGÃO", "CÓDIGO UNIDADE GESTORA"], axis = 1, inplace = True)

In [17]:
datos_brasil.columns

Index(['CÓDIGO ÓRGÃO SUPERIOR', 'NOME ÓRGÃO SUPERIOR', 'CÓDIGO ÓRGÃO',
       'NOME ÓRGÃO', 'CÓDIGO UNIDADE GESTORA', 'NOME UNIDADE GESTORA',
       'CATEGORIA ECONÔMICA', 'ORIGEM RECEITA', 'ESPÉCIE RECEITA',
       'DETALHAMENTO', 'VALOR PREVISTO ATUALIZADO', 'VALOR LANÇADO',
       'VALOR REALIZADO', 'PERCENTUAL REALIZADO', 'DATA LANÇAMENTO',
       'ANO EXERCÍCIO'],
      dtype='object')

In [18]:
fs.info_df(datos_brasil)

Tipo_dato  numero_registros  Numero_nulos  \
CÓDIGO ÓRGÃO SUPERIOR               int64           1026299             0   
NOME ÓRGÃO SUPERIOR                object           1026299             0   
CÓDIGO ÓRGÃO                        int64           1026299             0   
NOME ÓRGÃO                         object           1026299             0   
CÓDIGO UNIDADE GESTORA              int64           1026299             0   
NOME UNIDADE GESTORA               object           1026299             0   
CATEGORIA ECONÔMICA                object           1026299             0   
ORIGEM RECEITA                     object           1026299             0   
ESPÉCIE RECEITA                    object           1026299             0   
DETALHAMENTO                       object           1026299             0   
VALOR PREVISTO ATUALIZADO         float64           1026299             0   
VALOR LANÇADO                     float64           1026299             0   
VALOR REALIZADO                   float64           1026299             0   
PERCENTUAL REALIZADO              float64           1026299             0   
DATA LANÇAMENTO            datetime64[ns]           1025721           578   
ANO EXERCÍCIO                       int64           1026299             0   

                           %_nulos  
CÓDIGO ÓRGÃO SUPERIOR         0.00  
NOME ÓRGÃO SUPERIOR           0.00  
CÓDIGO ÓRGÃO                  0.00  
NOME ÓRGÃO                    0.00  
CÓDIGO UNIDADE GESTORA        0.00  
NOME UNIDADE GESTORA          0.00  
CATEGORIA ECONÔMICA           0.00  
ORIGEM RECEITA                0.00  
ESPÉCIE RECEITA               0.00  
DETALHAMENTO                  0.00  
VALOR PREVISTO ATUALIZADO     0.00  
VALOR LANÇADO                 0.00  
VALOR REALIZADO               0.00  
PERCENTUAL REALIZADO          0.00  
DATA LANÇAMENTO               0.06  
ANO EXERCÍCIO                 0.00

In [19]:
datos_brasil.duplicated().sum()

np.int64(0)

 ## Análisis descriptivo

## Análisis variables numéricas

Las columnas del dataframe nos proporcionan información sobre como se estructura la contabilidad, la balanza de pagos de Brasil. Realizaremos un estudio más profundo sobre su estructura contable, y la proveniencia de los ingresos de este país.

In [89]:
# Creamos una función que realiza las mismas funciones que el describe, pero con más estadísticos, solo para elementos numéricos
def estadisticos_numericas_df(dataframe, lista_columnas):
    valores_curtosis = []
    valores_asimetria = []
    valores_moda = []
    #creamos un bucle for para calcular los valores de la curtosis y la asimetría
    for columna in lista_columnas:
        if dataframe[columna].dtype == int or dataframe[columna].dtype == float:

            curtosis = round(dataframe[columna].kurtosis(), 2)
            asimetria = round(dataframe[columna].skew(),2)
            moda = dataframe[columna].mode()

            valores_curtosis.append(curtosis)
            valores_asimetria.append(asimetria)
            valores_moda.append(float(moda.iloc[0]))
            
        elif is_datetime(dataframe[columna]):
            valores_curtosis.append(np.nan)
            valores_asimetria.append(np.nan)
            valores_moda.append(float(moda.iloc[0]))
            
    estadísticos_generales = round(dataframe.describe().T, 2)
    estadísticos_generales["Curtosis"] = valores_curtosis
    estadísticos_generales["Coef_asimetria"] = valores_asimetria
    estadísticos_generales["Moda"] = valores_moda

    return estadísticos_generales

In [34]:
datos_brasil.describe().T

count                           mean  \
CÓDIGO ÓRGÃO SUPERIOR      1026299.0                   32050.523288   
CÓDIGO ÓRGÃO               1026299.0                   31435.161341   
CÓDIGO UNIDADE GESTORA     1026299.0                  231293.743317   
VALOR PREVISTO ATUALIZADO  1026299.0                27611421.049683   
VALOR LANÇADO              1026299.0                 5242806.106963   
VALOR REALIZADO            1026299.0                24980422.309138   
PERCENTUAL REALIZADO       1026299.0                      80.313812   
DATA LANÇAMENTO              1025721  2018-09-20 12:23:18.569786368   
ANO EXERCÍCIO              1026299.0                    2018.225555   

                                           min                  25%  \
CÓDIGO ÓRGÃO SUPERIOR                  20000.0              25000.0   
CÓDIGO ÓRGÃO                           20101.0              25201.0   
CÓDIGO UNIDADE GESTORA                110005.0             154003.0   
VALOR PREVISTO ATUALIZADO         -214773743.0                  0.0   
VALOR LANÇADO             -148347267246.910004                  0.0   
VALOR REALIZADO           -156285934188.029999              260.905   
PERCENTUAL REALIZADO                 -114552.0                  0.0   
DATA LANÇAMENTO            2013-12-31 00:00:00  2017-03-31 00:00:00   
ANO EXERCÍCIO                           2013.0               2017.0   

                                           50%                  75%  \
CÓDIGO ÓRGÃO SUPERIOR                  26000.0              39000.0   
CÓDIGO ÓRGÃO                           26298.0              36000.0   
CÓDIGO UNIDADE GESTORA                170013.0             250088.0   
VALOR PREVISTO ATUALIZADO                  0.0                  0.0   
VALOR LANÇADO                              0.0                  0.0   
VALOR REALIZADO                        3022.32            40853.855   
PERCENTUAL REALIZADO                       0.0                  0.0   
DATA LANÇAMENTO            2018-08-27 00:00:00  2020-02-12 00:00:00   
ANO EXERCÍCIO                           2018.0               2020.0   

                                           max                std  
CÓDIGO ÓRGÃO SUPERIOR                  81000.0       10612.547085  
CÓDIGO ÓRGÃO                           91214.0       10382.647574  
CÓDIGO UNIDADE GESTORA                913001.0      144178.420681  
VALOR PREVISTO ATUALIZADO      1603521711208.0  3393367175.127452  
VALOR LANÇADO              357160677863.969971   810883695.355811  
VALOR REALIZADO            771117711060.949951    1406892548.8405  
PERCENTUAL REALIZADO                72363772.0       71481.112902  
DATA LANÇAMENTO            2021-12-03 00:00:00                NaN  
ANO EXERCÍCIO                           2021.0           1.737966

In [57]:
datos_brasil["VALOR LANÇADO"].mode()

0    0.0
Name: VALOR LANÇADO, dtype: float64

In [59]:
float(datos_brasil["VALOR LANÇADO"].mode().iloc[0])

0.0

In [67]:
columnas_numericas2 = datos_brasil.select_dtypes(include = [np.number, "datetime"]).columns
columnas_numericas2

Index(['CÓDIGO ÓRGÃO SUPERIOR', 'CÓDIGO ÓRGÃO', 'CÓDIGO UNIDADE GESTORA',
       'VALOR PREVISTO ATUALIZADO', 'VALOR LANÇADO', 'VALOR REALIZADO',
       'PERCENTUAL REALIZADO', 'DATA LANÇAMENTO', 'ANO EXERCÍCIO'],
      dtype='object')

In [90]:
# Vamos a revisar los estadísticos de las variables numéricas de nuestro dataset.
estadisticos_numericas_df(datos_brasil, columnas_numericas2)

count                           mean  \
CÓDIGO ÓRGÃO SUPERIOR      1026299.0                   32050.523288   
CÓDIGO ÓRGÃO               1026299.0                   31435.161341   
CÓDIGO UNIDADE GESTORA     1026299.0                  231293.743317   
VALOR PREVISTO ATUALIZADO  1026299.0                27611421.049683   
VALOR LANÇADO              1026299.0                 5242806.106963   
VALOR REALIZADO            1026299.0                24980422.309138   
PERCENTUAL REALIZADO       1026299.0                      80.313812   
DATA LANÇAMENTO              1025721  2018-09-20 12:23:18.569786368   
ANO EXERCÍCIO              1026299.0                    2018.225555   

                                           min                  25%  \
CÓDIGO ÓRGÃO SUPERIOR                  20000.0              25000.0   
CÓDIGO ÓRGÃO                           20101.0              25201.0   
CÓDIGO UNIDADE GESTORA                110005.0             154003.0   
VALOR PREVISTO ATUALIZADO         -214773743.0                  0.0   
VALOR LANÇADO             -148347267246.910004                  0.0   
VALOR REALIZADO           -156285934188.029999              260.905   
PERCENTUAL REALIZADO                 -114552.0                  0.0   
DATA LANÇAMENTO            2013-12-31 00:00:00  2017-03-31 00:00:00   
ANO EXERCÍCIO                           2013.0               2017.0   

                                           50%                  75%  \
CÓDIGO ÓRGÃO SUPERIOR                  26000.0              39000.0   
CÓDIGO ÓRGÃO                           26298.0              36000.0   
CÓDIGO UNIDADE GESTORA                170013.0             250088.0   
VALOR PREVISTO ATUALIZADO                  0.0                  0.0   
VALOR LANÇADO                              0.0                  0.0   
VALOR REALIZADO                        3022.32            40853.855   
PERCENTUAL REALIZADO                       0.0                  0.0   
DATA LANÇAMENTO            2018-08-27 00:00:00  2020-02-12 00:00:00   
ANO EXERCÍCIO                           2018.0               2020.0   

                                           max                std    Curtosis  \
CÓDIGO ÓRGÃO SUPERIOR                  81000.0       10612.547085        0.26   
CÓDIGO ÓRGÃO                           91214.0       10382.647574        1.37   
CÓDIGO UNIDADE GESTORA                913001.0      144178.420681        4.57   
VALOR PREVISTO ATUALIZADO      1603521711208.0  3393367175.127452    88556.59   
VALOR LANÇADO              357160677863.969971   810883695.355811    80180.68   
VALOR REALIZADO            771117711060.949951    1406892548.8405   126151.22   
PERCENTUAL REALIZADO                72363772.0       71481.112902  1023400.74   
DATA LANÇAMENTO            2021-12-03 00:00:00                NaN         NaN   
ANO EXERCÍCIO                           2021.0           1.737966       -0.87   

                           Coef_asimetria      Moda  
CÓDIGO ÓRGÃO SUPERIOR                1.26   26000.0  
CÓDIGO ÓRGÃO                         1.50   25000.0  
CÓDIGO UNIDADE GESTORA               2.17  170013.0  
VALOR PREVISTO ATUALIZADO          264.11       0.0  
VALOR LANÇADO                      231.75       0.0  
VALOR REALIZADO                    290.10       0.0  
PERCENTUAL REALIZADO              1010.95       0.0  
DATA LANÇAMENTO                       NaN       0.0  
ANO EXERCÍCIO                        0.02    2016.0

### *Ingreso previsto actualizado*

El ingreso previsto actualizado medio de Brasil entre 2013 y 2021 se sitúa en los 27.611.421'05 $. La mediana de este valor se sitúa en los 0 dólares. Observamos que la media y la mediana no son valores cercanos, por lo que no vamos a considerar la media como un estadístico representativo de la muestra en este caso. 

La desviación típica se sitúa en un valor de 3.393.367.000 $, lo cual, es una variación considerablemente alta. Nos vamos a encontrar con una alta variabilidad en los ingresos previstos, ya que hay valores menores o iguales a cero y valores rondando el millón.

La asimetría presenta un valor de 264,11 y la media es superior a la mediana, por lo que estamos en un caso de asimetría positiva. Esto indica que en la distribución de los datos, nos vamos a encontrar con más valores inferiores. La curtosis tiene un valor de 88.556,59 lo que significa que la distribución de los datos tendrá un apuntamiento elevado. Hay más valores extremos en los datos. Al fin y al cabo, tenemos valores igual a cero, y valores en torno al millón, como hemos comentado anteriormente. 

El rango de ingresos se encuentra entre los -214.773.700 $ y los 1.603.522.000.000 dólares, lo cual es un rango extremadamente alto. Hemos de considerar que estamos ante los ingresos de un país en un rango de 9 años. Por otro lado, será necesario revisar porque tenemos ingresos negativos. La moda se sitúa en los 0 dólares. En cuanto a los cuartiles, nos encontramos que un 75 % de los ingresos se encuentran por debajo de los 0 dólares. 

 

En la tabla a continuación podemos ver los valores que se encuentran en la columna y el número de veces que se repite cada valor.

In [11]:
# Valores de la columna de `Valor previsto atualizado`.

print(f"Los ingresos previstos de brasil entre 2013 y 2021 han sido: {datos_brasil["VALOR PREVISTO ATUALIZADO"].sum().round(2)} \n")
valor_previsto_actualizado = pd.DataFrame(datos_brasil["VALOR PREVISTO ATUALIZADO"].value_counts())
valor_previsto_actualizado

Los ingresos previstos de brasil entre 2013 y 2021 han sido: 28337573811868.72 



count
VALOR PREVISTO ATUALIZADO         
0.0                        1006459
1000000.0                       22
300000.0                        15
10000000.0                      14
100000.0                        13
...                            ...
129273010.0                      1
82042.0                          1
407371.0                         1
7416.0                           1
16923750.0                       1

[18588 rows x 1 columns]

### *Ingresos registrados*

El ingreso registrado medio de Brasil entre 2013 y 2021 se sitúa en los 5.242.806'11 $. La mediana de este valor se sitúa en los 0 dólares. Observamos que la media y la mediana no son valores cercanos, por lo que no vamos a considerar la media como un estadístico representativo de la muestra en este caso. 

La desviación típica se sitúa en un valor de 810.883.700 $, lo cual, es una variación considerablemente alta. Nos vamos a encontrar con una alta variabilidad en los ingresos previstos, ya que hay valores menores e iguales a cero y valores rondando el millón.

La asimetría presenta un valor de 231,75 y la media es superior a la mediana, por lo que estamos en un caso de asimetría positiva. Esto indica que en la distribución de los datos, nos vamos a encontrar con más valores inferiores. La curtosis tiene un valor de 80.180,68 lo que significa que la distribución de los datos tendrá un apuntamiento elevado. Hay más valores extremos en los datos. 

El rango de ingresos se encuentra entre los -148.347.300.000 $ y los 357.160.700.000 dólares, lo cual es un rango extremadamente alto, aunque debemos tener en cuenta que estamos ante los ingresos de un país en un período temporal de 9 años. Por otro lado, será necesario revisar porque tenemos ingresos negativos. La moda se sitúa en los 0 dólares. En cuanto a los cuartiles, nos encontramos que un 75 % de los ingresos se encuentran por debajo de los 0 dólares. 

En la tabla a continuación podemos ver los valores que se encuentran en la columna y el número de veces que se repite cada valor.

In [12]:
# Valores de la columna de `Valor lanzado`.

print(f"Los ingresos recaudados registrados en Brasil entre 2013 y 2021 han sido: {datos_brasil["VALOR LANÇADO"].sum().round(2)} \n")
valor_previsto_actualizado = pd.DataFrame(datos_brasil["VALOR LANÇADO"].value_counts())
valor_previsto_actualizado

Los ingresos recaudados registrados en Brasil entre 2013 y 2021 han sido: 5380686664769.53 



count
VALOR LANÇADO         
 0.000000e+00  1019081
-1.000000e-02       32
 1.000000e-02       25
 1.100000e-01       10
 2.000000e-02        9
...                ...
 7.691600e+02        1
-9.852936e+04        1
 9.852936e+04        1
-4.151808e+04        1
 1.901645e+08        1

[6789 rows x 1 columns]

### *Ingresos recaudados*

El ingreso previsto actualizado medio de Brasil entre 2013 y 2021 se sitúa en los 24.980.422'31 $. La mediana de este valor se sitúa en los 3.022,32 dólares. Observamos que la media y la mediana no son valores cercanos, por lo que no vamos a considerar la media como un estadístico representativo de la muestra en este caso. 

La desviación típica se sitúa en un valor de 1.406.893.000 $, lo cual, es una variación respecto de la media considerablemente alta. Nos vamos a encontrar con una alta variabilidad en los ingresos previstos, ya que hay valores menores e iguales a cero y valores rondando el millón.

La asimetría presenta un valor de 290,10 y la media es superior a la mediana, por lo que estamos en un caso de asimetría positiva. Esto indica que en la distribución de los datos, nos vamos a encontrar con más valores inferiores. La curtosis tiene un valor de 126.151,22 lo que significa que la distribución de los datos tendrá un apuntamiento elevado. Hay más valores extremos en los datos. 

El rango de ingresos se encuentra entre los -156.285.900.000 $ y los 1.603.522.000.000 dólares, lo cual es un rango extremadamente alto, aunque debemos tener en cuenta que estamos ante los ingresos de un país en un período temporal de 9 años. Por otro lado, será necesario revisar porqué tenemos ingresos negativos. La moda se sitúa en los 0 dólares. En cuanto a los cuartiles, nos encontramos que un 75 % de los ingresos se encuentran por debajo de los 40.853,85 dólares, un valor considerablemente bajo.

En la tabla a continuación podemos ver los valores que se encuentran en la columna y el número de veces que se repite cada valor.

In [13]:
# Valores de la columna de `Valor recaudado`.

print(f"Los ingresos recaudados en Brasil entre 2013 y 2021 han sido: {datos_brasil["VALOR REALIZADO"].sum().round(2)} \n")
valor_previsto_actualizado = pd.DataFrame(datos_brasil["VALOR REALIZADO"].value_counts())
valor_previsto_actualizado

Los ingresos recaudados en Brasil entre 2013 y 2021 han sido: 25637382435446.0 



count
VALOR REALIZADO       
0.00             21007
100.00            3515
200.00            2878
20.00             2858
50.00             2701
...                ...
130276.15            1
324333.68            1
237194.40            1
224389.59            1
1297.13              1

[605783 rows x 1 columns]

 ### *Porcentaje recaudado respecto del previsto*

 El porcentaje de ingresos recaudados respecto de los ingresos previstos medio de Brasil entre 2013 y 2021 se sitúa en los 80,31 %. La mediana de este valor se sitúa en un 0 %. Observamos que la media y la mediana no son valores cercanos, por lo que no vamos a considerar la media como un estadístico representativo de la muestra en este caso. 

La desviación típica se sitúa en un valor de 71.481,11 %, lo cual, es un valor considerablemente alto. 

La asimetría presenta un valor de 1.010,95 y la media es superior a la mediana, por lo que estamos en un caso de asimetría positiva. Esto indica que en la distribución de los datos, nos vamos a encontrar con más valores inferiores. La curtosis tiene un valor de 1.023.400,74 lo que significa que la distribución de los datos tendrá un apuntamiento elevado. Hay más valores extremos en los datos. 

El rango se encuentra entre -114.552,00 % y 72.363.770 %, un rango extremadamente alto. En cuanto a los cuartiles, nos encontramos que un 75 % de los ingresos se encuentran por debajo del 0 %.

En la tabla a continuación podemos ver los valores que se encuentran en la columna y el número de veces que se repite cada valor.

In [14]:
# Valores de la columna de `percentual realizado`.

print(f"Los ingresos recaudados en Brasil entre 2013 y 2021 han sido: {datos_brasil["PERCENTUAL REALIZADO"].sum().round(2)} \n")
valor_previsto_actualizado = pd.DataFrame(datos_brasil["PERCENTUAL REALIZADO"].value_counts())
valor_previsto_actualizado

Los ingresos recaudados en Brasil entre 2013 y 2021 han sido: 82425985.0 



count
PERCENTUAL REALIZADO         
0.0                   1018808
1.0                       570
2.0                       218
3.0                       148
4.0                       112
...                       ...
501.0                       1
48270.0                     1
444.0                       1
10397.0                     1
7627.0                      1

[971 rows x 1 columns]

### *Fecha de registro de ejecución del ingreso*

In [15]:
datos_brasil["DATA LANÇAMENTO"].describe().reset_index()

index                DATA LANÇAMENTO
0  count                        1025721
1   mean  2018-09-20 12:23:18.569786368
2    min            2013-12-31 00:00:00
3    25%            2017-03-31 00:00:00
4    50%            2018-08-27 00:00:00
5    75%            2020-02-12 00:00:00
6    max            2021-12-03 00:00:00

In [ ]:
# Valores de la columna de `Data lancamento `.

print(f"Los ingresos de Brasil han sido registrados en las siguientes fechas: \n \n { datos_brasil["DATA LANÇAMENTO"].unique()}")

Los ingresos de Brasil han sido registrados en las siguientes fechas: 
 
 <DatetimeArray>
['2013-12-31 00:00:00', '2014-12-31 00:00:00', '2015-12-31 00:00:00',
 '2016-04-04 00:00:00', '2016-12-08 00:00:00', '2016-08-22 00:00:00',
 '2016-03-21 00:00:00', '2016-11-04 00:00:00', '2016-04-27 00:00:00',
 '2016-02-11 00:00:00',
 ...
 '2021-02-07 00:00:00', '2021-02-06 00:00:00', '2021-01-16 00:00:00',
 '2021-01-03 00:00:00', '2021-01-17 00:00:00', '2021-10-09 00:00:00',
 '2021-02-14 00:00:00', '2021-01-09 00:00:00', '2021-01-23 00:00:00',
 '2021-01-24 00:00:00']
Length: 2158, dtype: datetime64[ns]


## Análisis variables categóricas

Las columnas con variables categóricas nos proporcionan información sobre la proveniencia de los ingresos de Brasil, así como de la estructura gubernamental del país entre 2013 y 2021. A continuación realizamos un análisis de estas variables de forma detallada. 

In [17]:
# Sacamos los estadísticos de las variables de las columnas categóricas. 
datos_brasil.describe(include = "object").T

count unique  \
NOME ÓRGÃO SUPERIOR   1026299     25   
NOME ÓRGÃO            1026299    287   
NOME UNIDADE GESTORA  1026299    356   
CATEGORIA ECONÔMICA   1026299      5   
ORIGEM RECEITA        1026299     15   
ESPÉCIE RECEITA       1026299     63   
DETALHAMENTO          1026299   1886   

                                                                    top  \
NOME ÓRGÃO SUPERIOR                              Ministério da Educação   
NOME ÓRGÃO            Ministério da Economia - Unidades com vínculo ...   
NOME UNIDADE GESTORA            SETORIAL ORCAMENTARIA E FINANCEIRA / ME   
CATEGORIA ECONÔMICA                                  Receitas Correntes   
ORIGEM RECEITA                                Outras Receitas Correntes   
ESPÉCIE RECEITA            Serviços Administrativos e Comerciais Gerais   
DETALHAMENTO                SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC.   

                        freq  
NOME ÓRGÃO SUPERIOR   361891  
NOME ÓRGÃO            119811  
NOME UNIDADE GESTORA  108481  
CATEGORIA ECONÔMICA   979596  
ORIGEM RECEITA        322907  
ESPÉCIE RECEITA       278157  
DETALHAMENTO          159039

A continuación revisamos la información columna a columna.

### *Entidad gubernamental superior*

En Brasil, entre 2013 y 2021 nos encontramos con un total de 25 organismos superiores. El organismo que más veces se repite es el Ministerio de Educación, el cual ha ingresado un total de 221.589.353.642,78 $. Este valor se encuentra muy por encima de los ingresos medios. 

In [18]:
ingreso_entidad_superior = datos_brasil[datos_brasil["NOME ÓRGÃO SUPERIOR"] == "Ministério da Educação"]["VALOR REALIZADO"].sum().round(2)
ingreso_entidad_superior

np.float64(221589353642.78)

In [19]:
# Valores de la columna de `Organización superior`

print("Las organizaciones gubernamentales superiores de Brasil entre 2013 y 2021 son: \n ")
detalle_categoría_economica = pd.DataFrame(datos_brasil["NOME ÓRGÃO SUPERIOR"].value_counts())
detalle_categoría_economica["Peso categoría"] = round(((detalle_categoría_economica["count"]/datos_brasil.shape[0])*100), 2)
detalle_categoría_economica


Las organizaciones gubernamentales superiores de Brasil entre 2013 y 2021 son: 
 


count  Peso categoría
NOME ÓRGÃO SUPERIOR                                                  
Ministério da Educação                         361891           35.26
Ministério da Economia                         207574           20.23
Ministério da Defesa                            95490            9.30
Ministério da Agricultura, Pecuária e Abastec   46183            4.50
Ministério da Justiça e Segurança Pública       39980            3.90
Ministério da Infraestrutura                    37476            3.65
Ministério da Saúde                             33380            3.25
Ministério de Minas e Energia                   33014            3.22
Ministério da Ciência, Tecnologia, Inovações    32846            3.20
Ministério do Desenvolvimento Regional          32725            3.19
Ministério das Comunicações                     29485            2.87
Ministério do Meio Ambiente                     27151            2.65
Ministério do Turismo                           27096            2.64
Ministério da Cidadania                          5898            0.57
Presidência da República                         4976            0.48
Advocacia-Geral da União                         3899            0.38
Ministério das Relações Exteriores               3563            0.35
Ministério da Mulher, Família e Direitos Huma    1157            0.11
Ministério do Trabalho e Emprego                  685            0.07
Ministério do Desenvolvimento Agrário             666            0.06
Ministério da Pesca e Aquicultura                 373            0.04
Ministério das Mulheres, Igualdade Racial, da     285            0.03
Controladoria-Geral da União                      252            0.02
Ministério do Esporte                             147            0.01
Ministério da Previdência Social                  107            0.01

### *Entidad gubernamental específica*

En Brasil, entre 2013 y 2021 nos encontramos con un total de 297 organismos específicos. El organismo que más veces se repite es el Ministério da Economia - Unidades com vínculo direto, el cual ha ingresado un total de 208.88.345.811.550,4 $. Este valor se encuentra muy por encima de los ingresos medios. 

In [20]:
ingreso_entidad_específica = datos_brasil[datos_brasil["NOME ÓRGÃO"] == "Ministério da Economia - Unidades com vínculo direto"]["VALOR REALIZADO"].sum().round(2)
ingreso_entidad_específica

np.float64(20888345811550.4)

In [21]:
# Valores de la columna de `Nombre de organización`.

print("Las entidades gubernamentales específicas que nos encontramos en la adminsitración de Brasil son: \n ")
detalle_categoría_economica = pd.DataFrame(datos_brasil["NOME ÓRGÃO"].value_counts())
detalle_categoría_economica["Peso categoría"] = round(((detalle_categoría_economica["count"]/datos_brasil.shape[0])*100), 2)
detalle_categoría_economica

Las entidades gubernamentales específicas que nos encontramos en la adminsitración de Brasil son: 
 


count  Peso categoría
NOME ÓRGÃO                                                                
Ministério da Economia - Unidades com vínculo d...  119811           11.67
Fundo do Exército                                    25559            2.49
Comando da Marinha - Fundo Naval                     23734            2.31
Fundo Nacional de Cultura                            15955            1.55
Instituto Brasileiro do Meio Ambiente e dos Rec...   13745            1.34
...                                                    ...             ...
Hospital Cristo Redentor S.A.                            2            0.00
Agência de Desenvolvimento da Amazônia                   2            0.00
Companhia de Desenvolvimento de Barcarena                1            0.00
Departamento Nacional de Estradas de Rodagem             1            0.00
Hospital Fêmina S.A.                                     1            0.00

[287 rows x 2 columns]

### *Entidad gestora*

En Brasil, entre 2013 y 2021 nos encontramos con un total de 356 unidades gestoras. La unidad que más veces se repite es el SETORIAL ORCAMENTARIA E FINANCEIRA / ME	, el cual ha ingresado un total de 20.870.899.752.169,65 $. Este valor se encuentra muy por encima de los ingresos medios. 

In [22]:
ingreso_unidad_gestora = datos_brasil[datos_brasil["NOME UNIDADE GESTORA"] == "SETORIAL ORCAMENTARIA E FINANCEIRA / ME"]["VALOR REALIZADO"].sum().round(2)
ingreso_unidad_gestora

np.float64(20870899752169.65)

In [23]:
# Valores de la columna de `Nombre de organización gestora`.

print("Las organizaciones gestoras que nos encontramos en la adminsitración de Brasil son: \n ")
detalle_org_gestora = pd.DataFrame(datos_brasil["NOME UNIDADE GESTORA"].value_counts())
detalle_org_gestora["Peso categoría"] = round(((detalle_org_gestora["count"]/datos_brasil.shape[0])*100), 2)
detalle_org_gestora

Las organizaciones gestoras que nos encontramos en la adminsitración de Brasil son: 
 


count  Peso categoría
NOME UNIDADE GESTORA                                                 
SETORIAL ORCAMENTARIA E FINANCEIRA / ME        108481           10.57
FUNDO DO EXERCITO                               25469            2.48
DIRETORIA DE FINANCAS-SISTEMA PARA O PAIS/MM    23532            2.29
IBAMA - INST.BRASILEIRO MEIO AMBIENTE/MATRIZ    13698            1.33
EMBRAPA-SETORIAL                                13641            1.33
...                                               ...             ...
IBAMA - SUPERINTENDENCIA DE RORAIMA/RR              2            0.00
COORDENACAO DE ORCAMENTO E FINANCAS - COF/DPF       1            0.00
HOSPITAL FEMINA S/A                                 1            0.00
DEPARTAMENTO NAC. ESTRADAS E RODAGEM-EXTINCAO       1            0.00
COMPANHIA DE DESENVOLVIMENTO DE BARCARENA           1            0.00

[356 rows x 2 columns]

### *Clasificación económica de los ingresos*

En Brasil, entre 2013 y 2021 los ingresos se han clasificado en un total de 5 categorías económicas generales. La categoría que más ingresos ha recaudado son los ingresos corrientes, con un total de 12.909.343.128.909,44 $. 

In [24]:
ingresos_corrientes_totales = datos_brasil[datos_brasil["CATEGORIA ECONÔMICA"] == "Receitas Correntes"]["VALOR REALIZADO"].sum().round(2)
ingresos_corrientes_totales

np.float64(12909343128909.44)

In [25]:
# Valores de la columna de `Categoría económica`.

print("Los ingresos de Brasil se clasifican en las siguientes: \n ")
detalle_categoría_economica = pd.DataFrame(datos_brasil["CATEGORIA ECONÔMICA"].value_counts())
detalle_categoría_economica["Peso categoría"] = round(((detalle_categoría_economica["count"]/datos_brasil.shape[0])*100), 2)
detalle_categoría_economica

Los ingresos de Brasil se clasifican en las siguientes: 
 


count  Peso categoría
CATEGORIA ECONÔMICA                                              
Receitas Correntes                         979596           95.45
Receitas de Capital                         30116            2.93
Receitas Correntes - intra-orçamentárias    16235            1.58
Sem informação                                257            0.03
Receitas de Capital - intra-orçamentárias      95            0.01

### *Origen de los ingresos*

En Brasil, entre 2013 y 2021 los ingresos han provenido de un total de 15 categorías diferentes. La categoría que más ingresos ha recaudado es la de Otros ingresos corrientes, con un total de 495.575.920.557,84 $. 

In [26]:
otros_ingresos_corrientes_totales = datos_brasil[datos_brasil["ORIGEM RECEITA"] == "Outras Receitas Correntes"]["VALOR REALIZADO"].sum().round(2)
otros_ingresos_corrientes_totales

np.float64(495575920557.84)

In [27]:
# Valores de la columna de `Origen de ingresos`.

print("Los ingresos de Brasil provienen de las siguientes categorías: \n ")
detalle_origen_ingreso = pd.DataFrame(datos_brasil["ORIGEM RECEITA"].value_counts())
detalle_origen_ingreso["Peso categoría"] = round(((detalle_origen_ingreso["count"]/datos_brasil.shape[0])*100), 2)
detalle_origen_ingreso


Los ingresos de Brasil provienen de las siguientes categorías: 
 


count  Peso categoría
ORIGEM RECEITA                                                     
Outras Receitas Correntes                    322907           31.46
Receita de Serviços                          322319           31.41
Receita Patrimonial                          163171           15.90
Impostos, Taxas e Contribuições de Melhoria   78340            7.63
Contribuições                                 66781            6.51
Receita Industrial                            18857            1.84
Receita Agropecuária                          13642            1.33
Alienação de Bens                             12418            1.21
Amortizações de Empréstimos                   11318            1.10
Transferências Correntes                       6385            0.62
Operações de Crédito                           5044            0.49
Receitas Correntes - a classificar             3429            0.33
Outras Receitas de Capital                     1060            0.10
Transferências de Capital                       371            0.04
Sem informação                                  257            0.03

### *Tipo de ingreso*

En Brasil, entre 2013 y 2021 los ingresos han sido de un total de 63 tipos diferentes. La categoría que más ingresos ha recaudado es la de Otros ingresos corrientes, con un total de 32.059.641.128,12 $.

In [28]:
tipo_ingresos = datos_brasil[datos_brasil["ESPÉCIE RECEITA"] == "Serviços Administrativos e Comerciais Gerais"]["VALOR REALIZADO"].sum().round(2)
tipo_ingresos

np.float64(32059641128.12)

In [29]:
# Valores de la columna de `Especie de ingreso`.

print("Los ingresos de Brasil provienen de las siguientes categorías: \n ")
detalle_especie_ingreso = pd.DataFrame(datos_brasil["ESPÉCIE RECEITA"].value_counts())
detalle_especie_ingreso["Peso categoría"] = round(((detalle_especie_ingreso["count"]/datos_brasil.shape[0])*100), 2)
detalle_especie_ingreso

Los ingresos de Brasil provienen de las siguientes categorías: 
 


count  Peso categoría
ESPÉCIE RECEITA                                                      
Serviços Administrativos e Comerciais Gerais   278157           27.10
Indenizações, restituições e ressarcimentos    143577           13.99
Multas administrativas, contratuais e judicia  115276           11.23
Exploração do patrimônio imobiliário do Estad   88862            8.66
Taxas                                           62336            6.07
...                                               ...             ...
Remuneração das Disponibilidades do Tesouro N       3            0.00
Receitas decorrentes de compensações ao Regim       2            0.00
Demais receitas de capital                          2            0.00
Receita da dívida ativa                             1            0.00
Integralização do capital social                    1            0.00

[63 rows x 2 columns]

### *Detalle de tipo de ingreso*

En Brasil, entre 2013 y 2021 los ingresos han sido de un total de 18.886 tipos detallados diferentes. La categoría que más ingresos ha recaudado es la de SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC., con un total de 11.191.514.628,86 $.


In [30]:
tipo_ingresos_detallado = datos_brasil[datos_brasil["DETALHAMENTO"] == "SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC."]["VALOR REALIZADO"].sum().round(2)
tipo_ingresos_detallado

np.float64(11191514628.86)

In [31]:
# Valores de la columna de `Detalle adicional del tipo de ingreso`.
print("Los ingresos de Brasil provienen de las siguientes categorías en 2013: \n")
detalle_adicional_ingreso = pd.DataFrame(datos_brasil["DETALHAMENTO"].value_counts())
detalle_adicional_ingreso["Peso categoría"] = round(((detalle_adicional_ingreso["count"]/datos_brasil.shape[0])*100), 2)
detalle_adicional_ingreso

Los ingresos de Brasil provienen de las siguientes categorías en 2013: 



count  Peso categoría
DETALHAMENTO                                                         
SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC.    159039           15.50
INSCR.EM CONCURSOS E PROC.SELETIVOS-PRINCIPAL   74638            7.27
ALUGUEIS E ARRENDAMENTOS-PRINCIPAL              71040            6.92
MULTAS PREVISTAS EM LEGISL.ESPECIFICA-PRINC.    44580            4.34
RESTIT.DE DESPESAS DE EXERC.ANTERIORES-PRINC.   31513            3.07
...                                               ...             ...
SERV.REG.AN.CONTR.PROD.SUJ.A NORM.VIGIL.SANIT       1            0.00
Valores Mobiliários                                 1            0.00
ITR-MUNICIPIOS CONVENIADOS-DIVIDA ATIVA             1            0.00
OUTORGA DIR.USO DE RECURSOS HIDRICOS-DIV.AT.        1            0.00
RESTIT.REC.PRIMARIOS TRANSFERIDOS-PRINC.            1            0.00

[1886 rows x 2 columns]

### *Ejercicios analizados*

In [32]:
# Valores de la columna de `Año ejercicio`.

print(f"Los ingresos de Brasil han sido registrados en las siguientes fechas: \n \n { datos_brasil["ANO EXERCÍCIO"].unique()}")

Los ingresos de Brasil han sido registrados en las siguientes fechas: 
 
 [2013 2014 2015 2016 2017 2018 2019 2020 2021]
